core/src/gas.rs

```diff
@@ -301,47 +301,6 @@ pub trait CountersOwner {
     fn set_gas_left(&mut self, gas_left: GasLeft);
 }
 
-/// Provide functionality to safely make gas refund after charge,
-/// it checks, that it's not refunded more gas than has been charged.
-pub struct GasRefunder {
-    amount: u64,
-}
-
-/// [GasRefunder] error, in case somebody tries to refund more than was charged.
-#[derive(Debug, derive_more::Display)]
-#[display(fmt = "Trying to refund {charged}, but {refunded} was charged")]
-pub struct RefundError {
-    charged: u64,
-    refunded: u64,
-}
-
-impl GasRefunder {
```

Comment:
Gas refunder can be used in other cases. I think we must left it as is

Reply:
There is no any good reason to keep unsued code in the codebase. It contradicts to the YAGNI. One can restore it from the source control if they need it.